pennylane_lightning/lightning_qubit/_state_vector.py

```diff
@@ -0,0 +1,386 @@
+        """
+        state = self.state_vector
+
+        # Skip over identity operations instead of performing
+        # matrix multiplication with it.
+        for operation in operations:
+            name = operation.name
+            if name == "Identity":
+                continue
+            method = getattr(state, name, None)
+            wires = self.wires.indices(operation.wires)
+
+            if method is not None:  # apply specialized gate
+                inv = False
+                param = operation.parameters
+                method(wires, inv, param)
+            elif (
+                name[0:2] == "C(" or name == "ControlledQubitUnitary" or name == "MultiControlledX"
+            ):  # apply n-controlled gate
+                print("hi")
+                self._apply_lightning_controlled(operation)
+            else:  # apply gate as a matrix
+                # Inverse can be set to False since qml.matrix(operation) is already in
+                # inverted form
+                method = getattr(state, "applyMatrix")
+                try:
+                    method(qml.matrix(operation), wires, False)
+                except AttributeError:  # pragma: no cover
+                    # To support older versions of PL
+                    method(operation.matrix, wires, False)
+
+    def apply_operations(self, operations):
+        """Applies operations to the state vector."""
+        # State preparation is currently done in Python
+        if operations:  # make sure operations[0] exists
+            if isinstance(operations[0], StatePrep):
+                self._apply_state_vector(operations[0].parameters[0].copy(), operations[0].wires)
+                operations = operations[1:]
+            elif isinstance(operations[0], BasisState):
+                self._apply_basis_state(operations[0].parameters[0], operations[0].wires)
+                operations = operations[1:]
+
+        for operation in operations:
+            if isinstance(operation, (StatePrep, BasisState)):
+                raise DeviceError(
+                    f"Operation {operation.name} cannot be used after other "
+                    f"Operations have already been applied on a {self.short_name} device."
+                )
+
+        self.apply_lightning(operations)
+
+    def get_final_state(self, circuit: QuantumScript, debugger=None):
+        """
+        Get the final state that results from executing the given quantum script.
+
+        This is an internal function that will be called by the successor to ``lightning.qubit``.
+
+        Args:
+            circuit (QuantumScript): The single circuit to simulate
+            debugger (._Debugger): The debugger to use
+
+        Returns:
+            Tuple: A tuple containing the Lightning final state handler of the quantum script and
+                whether the state has a batch dimension.
+
+        """
+
+        circuit = circuit.map_to_standard_wires()
```

Comment:
Now done in `LightningQubit2.execute`.

Reply:
I decided to re-add it because with this we have full functionality for the class.
I also, think this is a small cost for a nice extra safety layer.